src/averageAges.js

```diff
@@ -1,61 +1,46 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+function calculateAverage(value, count) {
+  return Math.round(value / count * 100) / 100 || 0;
+}
+
+function calculateAverageAge(peopleCategory) {
+  const averageYearsValue = peopleCategory
+    .map(person => person.died - person.born)
+    .reduce((totalYears, age) => totalYears + age, 0);
+
+  return calculateAverage(averageYearsValue, peopleCategory.length);
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const mens = people.filter(men => (century
+    ? men.sex === 'm' && Math.ceil(men.died / 100) === century
+    : men.sex === 'm'));
+
+  return calculateAverageAge(mens);
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womans = people.filter(person => (withChildren
+    ? person.sex === 'f' && people.some(woman => person.name === woman.mother)
+    : person.sex === 'f'));
+
+  return calculateAverageAge(womans);
 }
 
-/**
- * Implement calculateAverageAgeDiff function.
- *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
- *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
- *
- * @param {object[]} people
- * @param {boolean} onlyWithSon - optional
- *
- * @return {number}
- */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childs = people.filter(child => (onlyWithSon
+    ? child.sex === 'm' && people.find(woman => woman.name === child.mother)
+    : people.find(woman => woman.name === child.mother)));
+
+  const childsAvgAge = childs
+    .reduce((total, child) => (
+      total + (child.born - people.find(woman => (
+        woman.name === child.mother)
+      ).born)
+    ), 0);
```

Comment:
Will be awesome to create a separate variable for the mother. It will be easier to read your code